### ISSUE TITLE
Niddler sometimes not auto-connecting

### ISSUE THREAD

**Issue by matejdro** [Chimerapps/niddler-ui#27]
I have noticed that Niddler will sometimes not auto connect after launching the app with "Run" button.

I have all 3 checkboxes in settings checked. Most of the time it seems to work, but sometimes it doesn't (Niddler stays disconnected after run. Connecting manually after that works normally).

I realize this is pretty vague, what can I do to collect more information on when this happens?

**Comment by NicolaChimerapps** [comment]
The automatic connect is based on some semi-heuristics gathered from a number of places.

I will need to add some logging in the plugin before we can get actual traces.

Some questions though to help with the assessment:

- Which version of IntelliJ/Android studio are you running?
- Are you trying to launch on an actual device, a simulator or an emulator?
- What kind of project are you running: android, flutter or dart?
- Do you observe this (more) when you press run or debug?

**Comment by matejdro** [comment]
* Android Studio 4.1, latest stable as of now
* Actual device
* Android native project
* Run. I almost never use debug option so I can't comment on that one.

**Comment by matejdro** [comment]
One possible cause for this could be unexpected disconnect. I had to reboot the phone while Niddler was connected. After that, auto connect did not work until I restarted Android Studio.

**Comment by matejdro** [comment]
In recent months, "sometimes", turned into "always" here :/. About 95% of the time, Niddler will not auto connect.

**Comment by NicolaVerbeeck** [comment]
Is your setup still the same?

**Comment by matejdro** [comment]
Yes, apart from updating to the latest Studio.

**Comment by NicolaVerbeeck** [comment]
Artic fox or chipmunk? I expect to be doing some native development again _soon_, this means I will have more time to look into this. Hang tight!

**Comment by matejdro** [comment]
I've tried both and they both have auto-connect issues.

**Comment by NicolaVerbeeck** [comment]
I just tested it again and could not see any issue on my end. Could you add: https://plugins.chimerapps.com/pluginUpload/updatePlugins.xml as a custom plugin repository and then remove and reinstall niddler (should be 2.13.1). In this repo I can more quickly test new things without waiting for intellij approval ;)

In that version I have added a project settings (under other-settings, niddler, per-project) that will some log data to the event-viewer. I am curious to see what it logs when it is not working for you

**Comment by matejdro** [comment]
Installed custom version, got this in the logs after launching:

```
07:16	Created process listener: isDebug: false, connect debugger: true, reuse session: true
07:16	Launch succeeded
```

Niddler did not auto-connect.

That is with

```
Android Studio Arctic Fox | 2020.3.1 Patch 4
Build #AI-203.7717.56.2031.7935034, built on November 21, 2021
Runtime version: 11.0.10+0-b96-7249189 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.15.12-1-manjaro
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 16
```

**Comment by NicolaVerbeeck** [comment]
Ok good, that tells me something. It seems the process listener which is scanning the app's output is not getting the server starting line that it needs to auto-connect.
In the output of the run window, do you see a log in the form of: `I/Niddler: Niddler Server running on 36829 [265dee][waitingForDebugger=false]` ?

**Comment by matejdro** [comment]
Yes, here it is:

![image](https://user-images.githubusercontent.com/507922/148170670-8e41669a-54e2-4685-bbf2-f986c03bc1bf.png)

**Comment by NicolaVerbeeck** [comment]
I mean in the 'run' section, not logcat (as we don't scan logcat).

![image](https://user-images.githubusercontent.com/7778415/148175310-05af3600-2937-4363-83d6-5ff2987f1f6b.png)

**Comment by matejdro** [comment]
No, I see bunch of logcat entries in that window, but not all of them (some logcat entries are there, some (including Nidler), are missing).

**Comment by matejdro** [comment]
Just want to mention that I've been experiencing logcat issues on my setup (https://issuetracker.google.com/issues/150558653), but I'm not sure if those two are connected (logcat did not stop working when I tested Niddler right now).

**Comment by NicolaVerbeeck** [comment]
I did some updates to use different heuristics (using pid from some obscure method) only to realise that I set pid to -1 in niddler android library. :/

I will need to create an update for those and then you could try with 2.13.2 which is currently on the custom repo

**Comment by matejdro** [comment]
Sorry for the long delay.

Auto connect seems to work now with Niddler-ui 2.14.0 and niddler lib 1.6.2. Interestingly enough, I still do not get any Niddler-related logs in the Run window, even with the extra logging option enabled.
